airbyte-integrations/bases/base-normalization/normalization/transform_catalog/transform.py

```diff
@@ -22,15 +22,270 @@
+                    result.update(found)
+            elif is_object(properties[key]["type"]):
+                found = find_properties_object(path=f"{path}.{key}", field=key, properties=properties[key])
+                result.update(found)
+    return result
+
+
+def process_node(path: str, json_col: str, name: str, properties: dict, from_table: str = "", previous="with ", inject_cols="") -> dict:
+    result = {}
+    if previous == "with ":
+        prefix = previous
+    else:
+        prefix = previous + ","
+    node_properties = extract_node_properties(path=path, json_col=json_col, properties=properties)
+    node_columns = ",\n    ".join([sql for sql in node_properties.values()])
+    # FIXME: use DBT macros to be cross_db compatible instead
+    hash_node_columns = (
+        "coalesce(cast("
+        + ' as string), ""),\n      coalesce(cast('.join([column for column in node_properties.keys()])
+        + ' as string), "")'
+    )
+    node_sql = f"""{prefix}
+{name}_node as (
+  select
+    {inject_cols}
+    {node_columns}
+  from {from_table}
+),
+{name}_with_id as (
+  select
+    *,
+    to_hex(md5(concat(
+      {hash_node_columns}
+    ))) as _{name}_hashid
+  from {name}_node
+)"""
+    # SQL Query for current node's basic properties
+    result[name] = node_sql + select_table(f"{name}_with_id")
+
+    children_columns = extract_nested_properties(path=path, field=name, properties=properties)
+    if children_columns:
+        for col in children_columns.keys():
+            child_col, join_child_table = json_extract_nested_property(path=path, json_col=json_col, name=col, definition=properties[col])
+            child_sql = f"""{prefix}
+{name}_node as (
+  select
+    {child_col},
+    {node_columns}
+  from {from_table}
+),
+{name}_with_id as (
+  select
+    to_hex(md5(concat(
+      {hash_node_columns}
+    ))) as _{name}_hashid,
+    {col}
+  from {name}_node
+  {join_child_table}
+)"""
+            if children_columns[col]:
+                children = process_node(
+                    path="$",
+                    json_col=col,
+                    name=f"{name}_{col}",
+                    properties=children_columns[col],
+                    from_table=f"{name}_with_id",
+                    previous=child_sql,
+                    inject_cols=f"_{name}_hashid as _{name}_foreign_hashid,",
+                )
+                result.update(children)
+            else:
+                # SQL Query for current node's basic properties
+                result[f"{name}_{col}"] = child_sql + select_table(
+                    f"{name}_with_id",
+                    columns=f"""
+  _{name}_hashid as _{name}_foreign_hashid,
+  {col}
+""",
+                )
+    return result
+
+
+def generate_dbt_model(catalog: dict, json_col: str, from_table: str) -> dict:
+    result = {}
+    for obj in catalog["streams"]:
+        name = "undefined"
+        if "name" in obj:
```

Comment:
in the airbyte catalog the `name` field is required. `stream` key isn't a thing. i think these may be artifacts of the jsonschema catalog?